pkg/clients/rds/rds.go

```diff
@@ -187,3 +191,11 @@ func IsErrorNotFound(err error) bool {
 	}
 	return errors.Is(err, ErrDBInstanceNotFound)
 }
+
+// errorConvert handle Server Error and covert error type
+func errorConvert(err error) error {
+	if e, ok := err.(*sdkErrors.ServerError); ok && e.ErrorCode() == ErrCodeInstanceNotFound {
```

Comment:
@hasheddan Most of the exception from SDK should be throw to corssplane-runtime. But if the error code is "InvalidDBInstanceId.NotFound" means the resource create by provide already out of sync. One of the case is the admin user manual remove the resource from aliyun cloud. 
And should covert the ServerError to  a Business Exception (ErrDBInstanceNotFound） and will let provider handle the resource status in right way. 

Otherwise， the Observe will alway throw exception:

https://github.com/crossplane/provider-alibaba/blob/f7d0617918064fec5fbdc91eeb5d0b4d677913ba/pkg/controller/database/rdsinstance.go#L155

or the resource is alway in deleting status.

https://github.com/crossplane/provider-alibaba/blob/f7d0617918064fec5fbdc91eeb5d0b4d677913ba/pkg/controller/database/rdsinstance.go#L245